sklearn/ensemble/tests/test_weight_boosting.py

```diff
@@ -505,6 +496,52 @@ def test_multidimensional_X():
     boost.predict(X)
 
 
+class ClassifierWithoutWeight(DummyClassifier):
+    """Classifier not supporting `sample_weight`."""
+    def fit(self, X, y):
+        return super().fit(X, y)  # pragma: no cover
+
+
+@pytest.mark.parametrize("algorithm", ['SAMME', 'SAMME.R'])
+def test_adaboostclassifier_without_sample_weight(algorithm):
+    X, y = iris.data, iris.target
+    base_estimator = ClassifierWithoutWeight()
+    clf = AdaBoostClassifier(
+        base_estimator=base_estimator, algorithm=algorithm
+    )
+    err_msg = ("{} doesn't support sample_weight"
+               .format(base_estimator.__class__.__name__))
+    with pytest.raises(ValueError, match=err_msg):
+        clf.fit(X, y)
+
+
+def test_adaboost_regressor_sample_weight():
```

Comment:
It' feels a little bit weird that you test the classifier for the no sample weight case and the regressor for the sample weight case, but why not.

Reply:
I found back what was the line of dev:

* `test_adaboosclassifier_***` covered some missing line in coverage.
* `test_adaboostregressor_***` add the regression test given in the issue. I should update the what's new. Classifier and regressor do not share the same `_boost` methods.